internal/program/program_test.go

```diff
@@ -55,4 +56,9 @@ func TestLogger(t *testing.T) {
 	logs, err := os.ReadFile(f.Name())
 	require.NoError(t, err)
 	require.Contains(t, string(logs), "logging to")
+
+	if runtime.GOOS == "windows" {
+		// https://github.com/docker/compose/issues/8186#issuecomment-814180124
+		t.Log("Workaround for containerd/console bug")
```

Comment:
Removing this causes this test to fail on windows with the errors
```go
=== FAIL: internal/program TestLogger (unknown)
failed to get console mode for stdout: The handle is invalid.
failed to get console mode for stderr: The handle is invalid.
```

This is a workaround which I found accidentally, trying to debug why this was occuring.
Tried vscode debugger first but the test succeeded, then tried the classic print debug and this seems to have fixed it.
Logging something using `t.Log()` in any line after line 53 i.e. `require.Error(t, p.Run())` seems to fix this.